crates/biome_cli/src/execute/mod.rs

```diff
@@ -362,22 +369,87 @@ pub fn execute_mode(
         };
         migrate::run(payload)
     } else {
-        let (summary_result, diagnostics) = traverse(&mode, &mut session, cli_options, paths)?;
+        let (summary_result, diagnostics) = traverse(&execution, &mut session, cli_options, paths)?;
         let console = session.app.console;
-        let mut reporter = ConsoleReporterBuilder::default()
-            .with_execution(&mode)
-            .with_cli_options(cli_options)
-            .with_diagnostics(diagnostics)
-            .with_summary(&summary_result)
-            .finish();
-
-        let mut visitor = ConsoleReporterVisitor(console);
-        report(
-            &mut reporter,
-            &mut visitor,
-            &mode,
-            cli_options,
-            &summary_result,
-        )
+        let errors = summary_result.errors;
+        let skipped = summary_result.skipped;
+        let processed = summary_result.changed + summary_result.unchanged;
+
+        let should_exit_on_warnings = summary_result.warnings > 0 && cli_options.error_on_warnings;
+
+        match execution.report_mode {
+            ReportMode::Terminal => {
+                let reporter = ConsoleReporter {
+                    summary: summary_result,
+                    diagnostics_payload: DiagnosticsPayload {
+                        verbose: cli_options.verbose,
+                        diagnostic_level: cli_options.diagnostic_level,
+                        diagnostics,
+                    },
+                    execution: execution.clone(),
+                };
+                reporter.write(&mut ConsoleReporterVisitor(console))?;
+            }
+            ReportMode::Json { pretty } => {
+                console.error(markup!{
+                    <Warn>"The "<Emphasis>"--json"</Emphasis>" option is "<Underline>"unstable/experimental"</Underline>" and its output might change between patches/minor releases."</Warn>
+                });
+                let reporter = JsonReporter {
+                    summary: summary_result,
+                    diagnostics: DiagnosticsPayload {
+                        verbose: cli_options.verbose,
+                        diagnostic_level: cli_options.diagnostic_level,
+                        diagnostics,
+                    },
+                    execution: execution.clone(),
+                };
+                let mut buffer = JsonReporterVisitor::new(summary_result);
+                reporter.write(&mut buffer)?;
+                if pretty {
+                    let content = serde_json::to_string(&buffer).map_err(|error| {
+                        CliDiagnostic::Report(ReportDiagnostic::Serialisation(
+                            SerdeJsonError::from(error),
+                        ))
+                    })?;
+                    let report_file = BiomePath::new("_report_output.json");
```

Comment:
I wanted to format the JSON file using the same options that the user defined, that's why I used the workspace. Although, this is a hazard and causes issues if a user has a `_report_output.json` file. 

Ideally, in the future, we want to create a virtual file that doesn't conflict with the files of the user.

Reply:
We could also add `biome` in the filename?